src/lib/pubkey/kyber/kyber_common/kyber_symmetric_primitives.h

```diff
@@ -16,9 +16,20 @@
 #include <memory>
 #include <tuple>
 #include <vector>
+#include <span>
 
 namespace Botan {
 
+class Kyber_XOF
+   {
+   public:
+      virtual ~Kyber_XOF() {}
+
+      virtual void set_position(const std::tuple<uint8_t, uint8_t>& matrix_position) = 0;
+
+      virtual void write_output(uint8_t out[], size_t len) = 0;
```

Comment:
Maybe use `std::span<uint8_t>` for this out param?

Reply:
Yeah, I was mentally modeling this after `write_keystream`, but no reason not to use span here.

Reply:
Done